src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -410,6 +410,14 @@ class Config:
         private = True
         process_pool = True
 
+    def locked_datasets(self):
+        return self.query([
+            ['encrypted', '=', True], ['OR', [
+                ['properties.truenas:about_to_lock.value', '=', 'yes'], ['key_loaded', '=', False]
```

Comment:
Why do we write `properties.truenas:about_to_lock.value` to persistent ZFS metadata, can't we have this stored in middleware RAM instead? This property has no sense after middleware restart.